src/c++/perf_analyzer/main.cc

```diff
@@ -1768,6 +1778,11 @@ main(int argc, char** argv)
         request_rate_range[SEARCH_RANGE::kSTEP], search_mode, summary);
   }
 
+  if (triton::perfanalyzer::IsMPIRun()) {
+    MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
Instead of flowing through the barrier, shouldn't we allow processes log results one by one to avoid mangling of logs?

Reply:
there needs to be a barrier after the profile function call so that no load managers stop early in case another process is still measuring

as far as "log mangling", this multi-model feature implemented here in PA is intended to be only used by MA for right now, which already hides PA output by default. there's no really a way to make an MPI run cleanly print output from multiple processes to the same stdout. `mpiexec` already has ways to separate the output into distinct files based on the rank--this is what i use in a test of this feature